Source/DafnyCore/AST/Grammar/ParseErrors.cs

```diff
@@ -407,18 +510,562 @@ declared identifier.
+    OneAction("replace with ':='", range, ":=", false),
+    OneAction("replace with ':-", range, ":-", false),
+    OneAction("replace with ':|'", range, ":|", false),
+});
+
+    Add(ErrorId.p_elephant_has_one_lhs,
+    @"
+Within a function, the `:-` operator is limited to a most one left-hand-side and exactly one-right-hand-side.
+", Remove(false, "remove extra LHSs"));
+
+    Add(ErrorId.p_elephant_has_one_rhs,
+    @"
+This error only occurs when using the elephant operator `:-` in conjunction with
+[failure-compatible types](../DafnyRef/DafnyRef#sec-failure-compatible-types)
+and in the context of a let-or-fail expression.
+
+In contrast to the let expression (`:=`), which allows multiple parallel initializations, 
+the let-or-fail expression (`:-`) is implemented to
+only allow at most a single left-hand-side and exactly one right-hand-side.
+", Remove(false, "remove extra RHSs"));
+
+    Add(ErrorId.p_set_comprehension_needs_term_expression,
+    @"
+A set comprehension (1) declares one or more variables, (2) possibly states some limits on those variables, 
+and (3) states an expression over those variables that are the values in the set.
+
+If there is no expression, then the expression is taken to be just the _one_ declared variable.
+For instance one could write `set b: bool`, which is equivalent to `set b: bool :: b` and would be the set of all `bool` values.
+Another example is `set x: nat | x < 5, which is equivalent to `set x: nat | x < 5:: x` and would be the set `{0, 1, 2, 3, 4}`.
+
+But if more than one variable is declared, then there is no natural implicit expression to fill in after the `::` if it is omitted, 
+so some expression is required. The failing example above, for example, might use the expression `x * y`, as in 
+`set x, y  | 0 <= x < y < 10 :: x * y`, or any other expression over `x` and `y`.
+", InsertAfter(" :: 0", "insert example expression"));
+
+    Add(ErrorId.p_invalid_name_after_dot,
+    @"
+This error message is not reachable in current Dafny.
+If it occurs, please report an internal bug (or obsolete documentation).
+");
+
+    Add(ErrorId.p_bad_number_format,
+    @"
+This error can only result from an internal bug in the Dafny parser.
+The parser recognizes a legitimate sequence of digits or sequence of hexdigits or
+a decimal number and then passes that string to a libary routine to create a BigInteger
+or BigDecimal. Given the parser logic, that parsing should never fail.
+");
+
+    Add(ErrorId.p_bad_hex_number_format,
+    @"
+This error can only result from an internal bug in the Dafny parser.
+The parser recognizes a legitimate sequence of digits or sequence of hexdigits or
+a decimal number and then passes that string to a libary routine to create a BigInteger
+or BigDecimal. Given the parser logic, that parsing should never fail.
+");
+
+    Add(ErrorId.p_bad_decimal_number_format,
+    @"
+This error can only result from an internal bug in the Dafny parser.
```

Comment:
Can you please same strings into the same variable?

Reply:
Instead made the explanations more specific -- they arise from three different places in the code, so it is helpful to use different errorIds